### PR TITLE
Adding pillar2 connector

### PR DESCRIPTION
**Pull request by NC-albur**
Added a connector to Pillar2 backend to hit the endpoint for passing through the UKTR data to ETMP

### CODE REVIEW

File: app/uk/gov/hmrc/pillar2submissionapi/controllers/actions/SubscriptionDataRetrievalAction.scala [hunk 1]
```diff
@@ -34,6 +34,7 @@ class SubscriptionDataRetrievalActionImpl @Inject() (
     with Logging {
 
   override protected def transform[A](request: IdentifierRequest[A]): Future[SubscriptionDataRequest[A]] = {
+    println(request.clientPillar2Id + " -> dslkfjdsljf;lksdjf;lk")
```
Comment (MoDweik) [inline app/uk/gov/hmrc/pillar2submissionapi/controllers/actions/SubscriptionDataRetrievalAction.scala:37]: please remove this

File: app/uk/gov/hmrc/pillar2submissionapi/controllers/UktrSubmissionController.scala
Comment (MoDweik) [inline app/uk/gov/hmrc/pillar2submissionapi/controllers/UktrSubmissionController.scala:1]: (file names) we're using all-caps for acronyms (see here https://github.com/hmrc/pillar2-stubs/pull/42)

File: app/uk/gov/hmrc/pillar2submissionapi/connectors/Pillar2Connector.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import uk.gov.hmrc.pillar2submissionapi.config.AppConfig
+import uk.gov.hmrc.pillar2submissionapi.models.uktrsubmissions.UktrSubmission
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class Pillar2Connector @Inject() (val config: AppConfig, val http: HttpClient)(implicit ec: ExecutionContext) extends Logging {
```
[the remark this answers is not in the document]
Comment (NC-albur) [reply app/uk/gov/hmrc/pillar2submissionapi/connectors/Pillar2Connector.scala:30]: So would UKTaxReturnConnector be preferable?

File: app/uk/gov/hmrc/pillar2submissionapi/models/uktrsubmissions/UktrSubmission.scala [hunk 3]
```diff
@@ -20,19 +20,52 @@ import play.api.libs.json._
 
 import java.time.LocalDate
 
-trait UktrSubmission {
+sealed trait UKTRSubmission {
```
Comment (ridow-hmrc) [inline app/uk/gov/hmrc/pillar2submissionapi/models/uktrsubmissions/UktrSubmission.scala:23]: I put this in here so we can ensure all match statements are exhaustive

File: app/uk/gov/hmrc/pillar2submissionapi/controllers/UktrSubmissionController.scala
Comment (MoDweik) [inline app/uk/gov/hmrc/pillar2submissionapi/controllers/UktrSubmissionController.scala:1]: acronym not all caps in file name in a few files

File: it/test/uk/gov/hmrc/pillar2submissionapi/connectors/SubscriptionConnectorSpec.scala [hunk 4]
```diff
@@ -16,119 +16,36 @@
...
-          .thenReturn(
-            Future.successful(Left(BadRequest))
-          )
+  "readSubscription" should "return json when the backend has returned 200 OK with data" in {
+    val subcriptionConntector = app.injector.instanceOf[SubscriptionConnector]
+    stubGet(s"$readSubscriptionPath/$plrReference", OK, subscriptionSuccess.toString)
+    val result = subcriptionConntector.readSubscription(plrReference).futureValue
+    result.isRight mustBe true
+    result.value mustBe subscriptionDataJson
+    println(server.listAllStubMappings().getMappings)
```
Comment (MoDweik) [inline it/test/uk/gov/hmrc/pillar2submissionapi/connectors/SubscriptionConnectorSpec.scala:35]: please remove this

**Review by MoDweik — DISMISSED** [review]

**Pull request merged**